checker/check_result.go

```diff
@@ -25,12 +26,44 @@ const MaxResultConfidence = 10
 // ErrorDemoninatorZero indicates the denominator for a proportional result is 0.
 var ErrorDemoninatorZero = errors.New("internal error: denominator is 0")
 
+// Types of details.
+const (
```

Comment:
FYI, a good way to emulate enums in Golang is to use custom types and `iota`. See example [here](https://github.com/ossf/scorecard/blob/main/checks/ci_tests.go#L27). You can avoid the validation logic also with this.

Reply:
Thanks for the tip! I updated the code using `iota`